src/query/storage/m3/storage.go

```diff
@@ -135,7 +136,26 @@ func (s *m3storage) FetchBlocks(
 		StepSize: query.Interval,
 	}
 
-	blocks, err := m3db.ConvertM3DBSeriesIterators(raw, bounds, opts)
+	enforcer := options.Enforcer
+	if enforcer == nil {
+		enforcer = cost.NoopChainedEnforcer()
+	}
+
+	// TODO: mutating this array breaks the abstraction a bit, but it's the least fussy way I can think of to do this
```

Comment:
@arnikola @nikunjgit how's this look? Avoids the pooling issues by just reusing the existing `SeriesIterator`.

@richardartoul @robskillington is mutating this array (the result of `Session.FetchTagged`) a terrible idea? If I don't, I think I need to do something to the effect of:

```
accountedIters := pool.Get(raw.Len())

for i, iter := range raw.Iters() {
   accountedIters.Set(i, NewAccountedSeriesIter(iter, ...)
}

pool.Put(raw)
```

which seems wasteful.